nuxeo-quota-core/src/main/java/org/nuxeo/ecm/quota/size/DocumentsSizeUpdater.java

```diff
@@ -65,42 +74,55 @@ public void computeInitialStatistics(CoreSession session, QuotaStatsInitialWork
             root = session.getDocument(new PathRef(path));
             query += " AND ecm:path STARTSWITH " + NXQL.escapeString(path);
         }
-        // reset on all documents
-        // this will force an update if the quota addon was installed and then removed
-        long count;
-        IterableQueryResult res = session.queryAndFetch(query, "NXQL");
-        try {
-            count = res.size();
-            log.debug("Start iteration on " + count + " items");
+        // get uuids to compute
+        List<String> uuids;
+        try (IterableQueryResult res = session.queryAndFetch(query, "NXQL")) {
+            uuids = new ArrayList<>((int) Long.min(res.size(), Integer.MAX_VALUE));
```

Comment:
But this is loading all uuids in memory, which doesn't scale. I thought the idea of the improvement was to use some kind of scrolling... `queryAndFetch` itself is an iterable and is supposed to avoid loading everything in memory

Reply:
Indeed, I will use scroll API which allow us to paginate at query level and have a better handling of transaction.